examples/data-objects/codemirror/src/index.ts

```diff
@@ -12,26 +12,29 @@ import { ContainerRuntime } from "@fluidframework/container-runtime";
 import { IFluidDataStoreFactory, FlushMode } from "@fluidframework/runtime-definitions";
 import {
     innerRequestHandler,
+    rootDataStoreRequestHandler,
     buildRuntimeRequestHandler,
 } from "@fluidframework/request-handler";
 import { defaultRouteRequestHandler } from "@fluidframework/aqueduct";
 
 import { fluidExport as smde } from "./codemirror";
 
+const defaultComponentId = "default";
+const defaultComponent = "@fluid-example/smde";
+
 class CodeMirrorFactory implements IRuntimeFactory {
     public get IRuntimeFactory() { return this; }
+    private readonly registry = new Map<string, Promise<IFluidDataStoreFactory>>([
+        ["@fluid-example/smde", Promise.resolve(smde)],
+    ]);
 
+    /**
+     * @deprecated Use instantiateFirstTime/instantiateFromExisting as appropriate
+     */
     public async instantiateRuntime(context: IContainerContext): Promise<IRuntime> {
-        const registry = new Map<string, Promise<IFluidDataStoreFactory>>([
-            ["@fluid-example/smde", Promise.resolve(smde)],
-        ]);
-
-        const defaultComponentId = "default";
-        const defaultComponent = "@fluid-example/smde";
-
         const runtime = await ContainerRuntime.load(
```

Comment:
It is beneficial to replace instantiateRuntime() implementation with 
```typescript
   if (context.existing) {
      return this.instantiateFromExisting(...);
   }
   return this.instantiateFirstTime(...);
```

Reply:
I'd actually consider having a base class that does it (need to look at rest of PR to see how useful it would be)

One other thing to consider - it's useful to have optional hasInitialized() method.
Take a look at packages\framework\aqueduct\src\data-objects\pureDataObject.ts and 4 initialization methods.

One way to achieve it - have different names on interface and such base class.
For example interface might have single instantiateRuntime(context: IContainerContext, existing: boolean) method.
But base class that we offer would implement it and break it into 4 init methods (all optional).
I do not like keeping same name as it removes forcing function (when we remove it from API), but these are the options on the table to consider.

Reply:
Any recommendation of where such a base class may reside? I have mixed feelings about putting it inside the `container-definitions` package and I'm not enthusiastic about wiring a new dependency in our around this code.

Reply:
container-definitions should definitely have no code. It should be in runtime-utils or similar package, i.e. it's at runtime layer, not at container layer. It is optional / helper thing, clients do not need to use it